### ISSUE TITLE
Replace MariaDB by PostgreSQL

### ISSUE THREAD

**Issue by erozqba** [savoirfairelinux/sous-chef#353]
I was trying to update the project to the version 1.10 of Django and I get a warning about the configuration of the MySQL server and a link to this part of the documentation: 

> WARNINGS:
> ?: (mysql.W002) MySQL Strict Mode is not set for database connection 'default'
>         HINT: MySQL's Strict Mode fixes many data integrity problems in MySQL, such as data truncation upon insertion, by escalating warnings into errors. It is strongly recommended you activate it. See: https://docs.djangoproject.com/en/1.10/ref/databases/#mysql-sql-mode

I really don't like the part "... data truncation upon insertion ..." but this was not the only thing that made me think in replacing MariaDB by PostgreSQL. We are storing JSON in the database and we are using an application for only that "django-annoying" but we have the possibility to use a JSONField of Django if we use PostgreSQL https://docs.djangoproject.com/en/1.10/ref/contrib/postgres/forms/#jsonfield 

I decided to try the project with PostgreSQL instead of MariaDB and I had an issue with SQLAlchemy, so I ask the question on Monday, august 8th, on the #feastproject IRC (there is already another issue #339 to talk about use SQLAlchemy or not.)

I would like to get your opinions about the advantages or disadvantages of using PostgreSQL or MariaDB.  Thanks in advance!

**Comment by rjacquemin** [comment]
FYI Postgresql also support GIS that could be useful for the geographical data. 

http://www.postgis.net
https://docs.djangoproject.com/en/1.10/ref/contrib/gis/install/postgis/

**Comment by erozqba** [comment]
@manumilou, @lamontfr, @maximest-pierre, @RignonNoel, @cormier, @madmath, @ssirois, @guillaumep, @rjacquemin 

Anyone with a good reason of why we don't have to change MariaDB by PostgreSQL ?

**Comment by guillaumep** [comment]
@erozqba If I understand your message, you are proposing to change from MariaDB to PostgreSQL. But your proposal seems to lack a solid reason to do so, such as performance issue, or incompatibility with Django, or development of a new feature requiring PostgreSQL.

If I have a vote, I would vote against a switch, especially this late in the project.
If it is not broken, do not fix it.

**Comment by madmath** [comment]
What Guillaume says is reasonable. If there is one day a special feature that requires PostgreSQL (GIS stuff one day?),  it will make sense to migrate. For now the application is pretty simple and works equally well, unless you argue otherwise...

**Comment by manumilou** [comment]
I agree with both @madmath and @guillaumep . I don't think we have enough incentives to migrate to PostgreSQL. 

The main thing I see that could be interesting for us right now is the JSON fields support in PostgreSQL, which would allow us to get rid of the django-annoying dependency. But we have easier way to do that.

**Comment by erozqba** [comment]
I respectfully disagree with you (@guillaumep, @madmath, @manumilou), because the only reason that you point out is "We don't need to do it right now" and not what is the best option for the project now and in the future.

I think that MariaDB is a great DBMS and I have used it a lot more than PostgreSQL.

But for this project, I think PostgreSQL is a better choice because we need to store JSON right now and in the future if we need to add a feature like "GIS" we don't need to do a painful data migration. With PostgreSQL, all the logic that is handled in python to work with the JSON values will be easier and we will replace a custom Django app (django-annoying) by a field that is in Django itself. 

We are working on a project that is not deployed in production yet and change to PostgreSQL right now will be a lot easier than later in the future. 

What is the point in not use a tool (PostgreSQL) that should make the development process easier and with fewer custom dependencies because we have already used other tool (MySQL) and added the dependency (django-annoying) ? 

What is the solid reason that has made us use MariaDB in the first place that we need a solid reason to replace it? 

Even when I read the differents comparison in the web, in general, PostgreSQL is less popular but better than MariaDB.
